### PR TITLE
Add exercise perfect-numbers

### PR DESCRIPTION
**Pull request by komyg**
Hi, 

I would like to contribute with the perfect-numbers exercise implementation.

### CODE REVIEW

**Comment by matthewmorgan** [comment]
Right on!

We'll want this one to be as close to idiomatic ES6 as possible.  Please go ahead and implement a solution, and I can help you iterate it to ES6 code.

**Comment by komyg** [comment]
Hi @matthewmorgan, I believe I am all done, please tell me what you think.

File: exercises/perfect-numbers/example.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    // Check if the input is valid
+    if (number <= 0) {
+      return 'Classification is only possible for natural numbers.';
+    }
+
+    // Factorize the current number.
+    const divsArray = this.getDivisors(number);
+
+    // Sum the factors.
+    sum = 0;
```
Comment (matthewmorgan) [inline exercises/perfect-numbers/example.js]: No need for this initial assignment.  You can simply assign `sum` to the result of `reduce`.

File: exercises/perfect-numbers/example.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    // Calculate the divisors up the the half of the number + 1
+    for (i = 2; i <= number / 2; i++) {
+
+      if (number % i === 0) {
+        divs.push(i);
+  		}
+    }
+
+    return divs;
+  }
```
Comment (matthewmorgan) [inline exercises/perfect-numbers/example.js]: This looks good.  I was going to suggest an ES6-y approach using [generators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators) but we don't have the environment set up for that yet!

File: exercises/perfect-numbers/example.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+   * Calculate all the divisors for a given number.
+   * @param {number} number - natural number.
+   * @returns {array} array that contains the divisors for a given number NOT including the number itself.
+   */
+  getDivisors(number) {
+
+    let i;
+    let divs = new Array();
+
+    // Accepts only natura numbers greater than 1.
```
Comment (matthewmorgan) [inline exercises/perfect-numbers/example.js]: A typo here, should be `natural`!

**Comment by matthewmorgan** [comment]
💯 !!  Thanks again.  Nice work.

**Comment by komyg** [comment]
Happy to help!

**Pull request merged**